_posts/2020-01-08-#14053.md

```diff
@@ -0,0 +1,93 @@
+  - [Add address-based index by rnicoll](https://github.com/bitcoin/bitcoin/pull/6835)
+
+- Address indexes are inherently unscalable since they grow linearly with the
+  size of the block chain. Some contributors worry that offering an address
+  index in Bitcoin Core may encourage businesses to build Bitcoin services on
+  infrastructure that can't scale as the size of the block chain increases.
+  Pieter Wuille expresses this concern in the post from his PR: "I hate making it
+  easy to build infrastructure that relies on a fully-indexed blockchain (which
+  shouldn't be necessary), as it may remove the incentive to build more scalable
+  solutions. On the other hand, in cases where the alternative is relying on a
+  trusted third party, this approach is certainly preferable, and would allow an
+  RPC-based blockexplorer, for example."
+
+- For more discussion on the merits of adding an address index to Bitcoin Core,
+  see the discussion in the [13 Dec 2018 Bitcoin Core IRC
+  meeting](http://www.erisian.com.au/meetbot/bitcoin-core-dev/2018/bitcoin-core-dev.2018-12-13-19.00.log.html#l-271).
+
+- This latest attempt to add an address index builds on the new auxiliary
+  indexes base class introduced in [PR
+  13243](https://github.com/bitcoin/bitcoin/pull/13243). The auxiliary index
+  infrastructure runs asynchronously and uses the CValidationInterface to
+  receive new block data. That means adding new indexes should have minimal
+  impact on performance and low risk of introducing a bug that impacts mainline
+  functionality.
+
+- The new address index code and tests (`index/addrindex.cpp`,
+  `index/addrindex.h`, `test/addrindex_tests.cpp`) are similar to the
+  transcation index code and tests (`index/txindex.cpp`, `index/txindex.h`,
+  `test/txindex_tests.cpp`). Comparing them by eye or using a diff tool may be
+  helpful to understand what the new code is doing.
+
+## Questions
+
+1. Do you think Bitcoin Core should include an address index? What are the
+   arguments for and against?
+
+2. Did you review the PR? [Concept ACK, approach ACK, ACK \<commit\>, or
+  NACK?](https://github.com/jonatack/bitcoin-development/blob/master/how-to-review-bitcoin-core-prs.md#peer-review)&nbsp;
+  Don't forget to put your PR review on GitHub or [ask
+  questions](https://github.com/jonatack/bitcoin-development/blob/master/how-to-review-bitcoin-core-prs.md#ask-questions).
+
+3. What steps did you take to review and test this PR? Did you try to create an
+   address index over the full mainnet block chain?
+
+4. Did you review the unit and functional tests? Are there any additional test
+   cases that you'd like to see added?
+
+5. The address index is stored in
+  [LevelDB](https://github.com/bitcoin-core/leveldb) which is a key-value
+  store. What is used as the key, and what is used as the value?
+
+6. Where is murmurhash3 used in this PR? How are hash collisions resolved?
+   Where else is murmurhash3 used in the Bitcoin Core codebase?
```

Comment:
perhaps s/murmurhash3/`MurmurHash3`/ in these 2 lines as it's mostly in the latter form in the codebase afaict

Reply:
done